src/couch_file.erl

```diff
@@ -252,14 +252,23 @@ deleted_filename_suffix() ->
         [Y, Mon, D, H, Min, S, couch_uuids:random()]).
 
 nuke_dir(RootDelDir, Dir) ->
+    case config:get_boolean("couchdb", "rename_on_delete", false) of
+        true -> couch_server:delete_file(RootDelDir, Dir, []);
+        false -> delete_dir(RootDelDir, Dir)
+    end.
+
+delete_dir(RootDelDir, Dir) ->
     FoldFun = fun(File) ->
         Path = Dir ++ "/" ++ File,
         case filelib:is_dir(Path) of
             true ->
-                ok = nuke_dir(RootDelDir, Path),
+                ok = delete_dir(RootDelDir, Path),
                 file:del_dir(Path);
             false ->
-                delete(RootDelDir, Path, false)
+                UUID = couch_uuids:random(),
+                TmpFile = filename:join([RootDelDir, ".delete", UUID]),
+                file:rename(Path, TmpFile),
+                file:delete(TmpFile)
```

Comment:
I agree. The rename and delete isn't very useful here. I'd do what @rnewson suggests and just rename in nuke_dir, and then do a similar test as above in couch_file:delete that if rename_on_delete is true then we just don't do the recursion similar to what exists currently.